core/domain/story_domain.py

```diff
@@ -182,6 +211,70 @@ def create_default_story_node(cls, node_id):
         """
         return cls(node_id, [], [], [], '', None)
 
+    def validate(self):
+        """Validates various properties of the story node.
+
+        Raises:
+            ValidationError: One or more attributes of the story node are
+            invalid.
+        """
+        if self.exploration_id:
+            if not isinstance(self.exploration_id, basestring):
+                raise utils.ValidationError(
+                    'Expected exploration ID to be a string, received %s' %
+                    self.exploration_id)
+
+        if not isinstance(self.outline, basestring):
```

Comment:
Just remembered something -- for any html strings (in any of the new domain objects) please make sure to pass them through html_cleaner.clean() on init(). This is important for security. See exp_domain.py for examples.